Spec/Auth.swift

```diff
@@ -1232,6 +1233,107 @@ class Auth : QuickSpec {
                         done()
                     }
                 }
+                
+                tokenParams.clientId = "newClientId"
+                tokenParams.ttl = 2000
+                tokenParams.capability = "{ \"test:*\":[\"test\"] }"
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
+                        expect(error).to(beNil())
+                        guard let tokenRequest = tokenRequest else {
+                            XCTFail("tokenRequest is nil"); done(); return
+                        }
+                        expect(tokenRequest.clientId).to(equal("newClientId"))
+                        expect(tokenRequest.ttl).to(equal(2000))
+                        expect(tokenRequest.capability).to(equal("{ \"test:*\":[\"test\"] }"))
+                        done()
+                    }
+                }
+                
+                tokenParams.clientId = nil
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
+                        expect(error).to(beNil())
+                        guard let tokenRequest = tokenRequest else {
+                            XCTFail("TokenDetails is nil"); done(); return
+                        }
+                        expect(tokenRequest.clientId).to(beNil())
+                        done()
+                    }
+                }
+            }
+            
+            it("should override defaults if `AuthOptions` provided") {
+                var currentTokenRequest: ARTTokenRequest? = nil
+                var secondCallbackCalled = false
+                
+                let defaultOptions = AblyTests.commonAppSetup()
+                defaultOptions.authCallback = { tokenParams, completion in
+                    completion(currentTokenRequest!, nil)
+                }
+                
+                let rest = ARTRest(options: defaultOptions)
+                rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                    currentTokenRequest = tokenRequest
+                })
+                expect(currentTokenRequest).toEventuallyNot(beNil(), timeout: testTimeout)
+                
+                let options = ARTAuthOptions()
+                options.authCallback = { tokenParams, completion in
+                    secondCallbackCalled = true
+                    completion(currentTokenRequest!, nil)
+                }
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(nil, options: options) { tokenDetails, error in
```

Comment:
Shouldn't we testing the `createTokenRequest` (spec `(RSA9) Auth#createTokenRequest`)?